src/popup/components/AccountDetailsBase.vue

```diff
@@ -0,0 +1,228 @@
+<template>
+  <div
+    ref="accountDetailsElem"
+    class="account-details"
+  >
+    <div class="account-info-wrapper">
+      <slot name="account-info" />
+
+      <BtnClose
+        class="close-button"
+        :to="{ name: homeRouteName }"
+      />
+    </div>
+    <div>
+      <slot name="balance" />
+
+      <div class="buttons">
+        <slot name="buttons" />
+      </div>
+
+      <div class="header">
+        <slot name="navigation" />
+
+        <TransactionAndTokenFilter
+          :key="routeName"
+          :show-filters="showFilters"
+        />
+      </div>
+
+      <div class="tabs-content">
+        <transition
+          name="fade-transition"
+          mode="out-in"
+        >
+          <RouterView />
+        </transition>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import {
+  computed,
+  defineComponent,
+  onBeforeUnmount,
+  onMounted,
+  ref,
+  watch,
+} from '@vue/composition-api';
+import { debounce } from 'lodash-es';
+import {
+  EXTENSION_HEIGHT,
+} from '../utils';
+import {
+  useTransactionAndTokenFilter,
+  useUi,
+} from '../../composables';
+import { IS_CORDOVA } from '../../lib/environment';
+
+import BtnClose from './buttons/BtnClose.vue';
+import TransactionAndTokenFilter from './TransactionAndTokenFilter.vue';
+
+export default defineComponent({
+  name: 'AccountDetailsBase',
+  components: {
+    TransactionAndTokenFilter,
+    BtnClose,
+  },
+  setup(props, { root }) {
+    const ACCOUNT_INFO_HEIGHT = 120;
+    const BALANCE_AND_ACTIONS_HEIGHT = 280;
+    const accountDetailsElem = ref<HTMLElement>();
+    const appInnerScrollTop = ref<number>(0);
+    const clientHeight = ref<number>(0);
+    const initialClientHeight = ref<number>(EXTENSION_HEIGHT);
+
+    const { resetFilter } = useTransactionAndTokenFilter();
+
+    const { homeRouteName } = useUi({ store: root.$store });
+
+    const appInnerElem = computed<HTMLElement | null | undefined>(
+      () => accountDetailsElem.value?.parentElement,
+    );
+
+    const routeName = computed(() => root.$route.name);
+
+    const showFilters = computed<boolean>(() => (
+      clientHeight.value > initialClientHeight.value
+        && appInnerScrollTop.value >= ACCOUNT_INFO_HEIGHT
+    ));
+
+    const resizeObserver = new ResizeObserver(debounce((entries) => {
+      if (!(Array.isArray(entries) && entries.length)) {
+        return;
+      }
+      const newClientHeight = entries[0].target.clientHeight;
+      if (
+        newClientHeight && (
+          (
+            clientHeight.value + BALANCE_AND_ACTIONS_HEIGHT + ACCOUNT_INFO_HEIGHT
+          ) < newClientHeight
+              || newClientHeight <= initialClientHeight.value
+        )
```

Comment:
This condition looks messed up.
1. In case of more than one `and` the `&&` signs should be put in next line.
2. Long comparisions looks better when stored in constants.